src/workers/notifications/articleUpvoteMilestone.ts

```diff
@@ -40,7 +42,24 @@ const worker: NotificationWorker = {
       upvoters,
       upvotes: post.upvotes,
     };
-    return users.map((userId) => ({
+    if (source.type !== SourceType.Squad) {
+      return users.map((userId) => ({
+        type: 'article_upvote_milestone',
+        ctx: { ...ctx, userId },
+      }));
+    }
+
+    const members = await con.getRepository(SourceMember).findBy({
+      userId: In(users),
+      sourceId: source.id,
+      role: Not(SourceMemberRoles.Blocked),
+    });
+
+    if (!members.length) {
+      return;
+    }
+
+    return members.map(({ userId }) => ({
```

Comment:
Maybe do all of the squad logic inside `source.type === SourceType.Squad`, since squad source is specific permission maybe better to early return from there and then on the bottom the default action stays the same. Similarly how you did it in `src/workers/notifications/commentUpvoteMilestone.ts`

Reply:
Agreed that would be cleaner.